pages/instructorAssessmentGroups/instructorAssessmentGroups.ejs

```diff
@@ -0,0 +1,434 @@
+            data-target="#configModal">
+            <i class="fa fa-cog" aria-hidden="true"></i> Config
+          </button>
+          <button type="button" class="btn btn-sm btn-danger" tabindex="0" data-toggle="modal"
+            data-target="#deleteAllGroupsModal">
+            <i class="fa fa-times" aria-hidden="true"></i> Delete all groups
+          </button>
+        </div>
+      </div>
+      <div class="container-fluid">
+        <div class="row">
+          <div class="col-sm bg-light py-4 border" align="center">
+            <button type="button" class="btn btn-primary text-nowrap" data-toggle="modal"
+              data-target="#uploadAssessmentGroupsModal">
+              <i class="fas fa-upload" aria-hidden="true"></i> Upload
+            </button>
+            <div class="mt-2">
+              Upload a CSV format file for group setting.
+            </div>
+          </div>
+          <div class="col-sm bg-light py-4 border" align="center">
+            <button type="button" class="btn btn-primary text-nowrap" data-toggle="modal"
+              data-target="#autoAssessmentGroupsModal">
+              <i class="fas fa-robot" aria-hidden="true"></i> Auto
+            </button>
+            <div class="mt-2">
+              Quick group students into teams by system.
+            </div>
+          </div>
+          <div class="col-sm bg-light py-4 border" align="center">
+            <button type="button" class="btn btn-primary text-nowrap" data-toggle="modal"
+              data-target="#copyAssessmentGroupsModal">
+              <i class="fas fa-copy" aria-hidden="true"></i> Copy
+            </button>
+            <div class="mt-2">
+              Copy the group setting from another assessment.
+            </div>
+          </div>
+        </div>
+      </div>
+      <div class="table-responsive">
+        <table id="usersTable" class="table table-sm table-hover tablesorter">
+          <thead>
+            <tr>
+              <th>GID</th>
+              <th>Name</th>
+              <th>Size</th>
+              <th class="text-center">Group Members (UIDs)</th>
+              <th class="sorter-false"></th>
+            </tr>
+          </thead>
+          <tbody>
+            <% groups_rows.forEach(function(row, iRow) { %>
+            <tr>
+              <td><%= row.gid %></td>
+              <td><%= row.groupname %></td>
+              <td class="text-center"><%= row.num %></td>
+              <td class="text-center"><small><%= row.uids %></small></td>
+              <td class="text-center">
+                <div class="dropdown">
+                  <button type="button" class="btn btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true"
+                    aria-expanded="false">
+                    Action <span class="caret"></span>
+                  </button>
+                  <div id="row<%= iRow %>PopoverAdd" tabindex="0" data-toggle="popover" data-trigger="manual"
```

Comment:
Consider formatting with each attribute on a new line for improved readability. See https://github.com/PrairieLearn/PrairieLearn/blob/master/pages/partials/examQuestionHelpAwardedPoints.ejs for an example.